### PR TITLE
Added implementation,testcases for validate text and validate attribute

### PR DESCRIPTION
**Pull request by vidyaravind**

### CODE REVIEW

File: lib/locreator.js [hunk 1]
```diff
@@ -75,6 +75,12 @@ Locreator.prototype.addStarMethods = function locreator(locatorId, _locator, par
   locatorObject[locatorId + 'OptionValue'] = function (optionValue) {
     return drivex.selectByOptionValue(locator, optionValue, parentWebElement);
   };
+  locatorObject[locatorId + 'ValidateText'] = function (ValidateText) {
```
Comment (grawk) [inline lib/locreator.js]: What about calling the method "TextEquals" instead? Trying to shorten the name if possible

File: lib/locreator.js [hunk 2]
```diff
@@ -75,6 +75,12 @@ Locreator.prototype.addStarMethods = function locreator(locatorId, _locator, par
...
+    return drivex.validateText(locator, parentWebElement,ValidateText) ;
+  };
+  locatorObject[locatorId + 'ValidateAttributeValue'] = function (attribute,validateText) {
```
Comment (grawk) [inline lib/locreator.js]: What about calling the method "AttributeEquals" or "AttrEquals"? Trying to shorten the name. @nikulkarni what do you think?

File: test/methods.js [hunk 3]
```diff
@@ -124,7 +124,32 @@ describe('nemo-view @methods@', function () {
       done();
     }, util.doneError(done));
   });
-
+  it('should return true using @ValidateText@Positive@ method', function (done) {
+    nemo.view.simple.pageHeaderValidateText('Sample form stuff').then(function (actual) {
```
Comment (grawk) [inline test/methods.js]: naming the argument `actual` doesn't indicate it is a boolean value. Please change to something like `isEqual`

File: test/methods.js [hunk 4]
```diff
@@ -124,7 +124,32 @@ describe('nemo-view @methods@', function () {
...
+      assert.equal(actual, true);
+      done();
+    }, util.doneError(done));
+  });
+  it('should return false using @ValidateText@negative@ method', function (done) {
+    nemo.view.simple.pageHeaderValidateText('form stuff').then(function () {
+      done(new Error('couldn\'t find text: form stuff'));
```
Comment (grawk) [inline test/methods.js]: The error here would be that the method resolved it's promise when it shouldn't have. I.e. the message should be "resolved promise as true when we should have rejected it" or something like that

File: test/methods.js [hunk 5]
```diff
@@ -124,7 +124,32 @@ describe('nemo-view @methods@', function () {
...
+    }, function (err) {
+      done();
+    });
+  });
+  it('should return true using @AttributeValue@Positive@ method', function (done) {
+    nemo.view.simple.buttonLabelValidateAttributeValue('value','Go foo').then(function (actual) {
```
Comment (grawk) [inline test/methods.js]: the variable name `actual` doesn't indicate a boolean. please use something like `isEqual`

File: test/methods.js [hunk 6]
```diff
@@ -124,7 +124,32 @@ describe('nemo-view @methods@', function () {
...
+      assert.equal(actual, true);
+      done();
+    }, util.doneError(done));
+  });
+  it('should return false using @AttributeValue@negative@ method', function (done) {
+    nemo.view.simple.buttonLabelValidateAttributeValue('value','foo').then(function () {
+      done(new Error('couldn\'t find text: foo'));
```
Comment (grawk) [inline test/methods.js]: error message would be misleading here. the error is that the promise resolved when it should have rejected

File: test/methods.js [hunk 7]
```diff
@@ -124,7 +124,32 @@ describe('nemo-view @methods@', function () {
...
+    });
+  });
+  it('should return true using @AttributeValue@Positive@ method', function (done) {
+    nemo.view.simple.buttonLabelValidateAttributeValue('value','Go foo').then(function (actual) {
+      assert.equal(actual, true);
+      done();
+    }, util.doneError(done));
+  });
+  it('should return false using @AttributeValue@negative@ method', function (done) {
+    nemo.view.simple.buttonLabelValidateAttributeValue('value','foo').then(function () {
```
Comment (grawk) [inline test/methods.js]: whitespace, please put a space between a comma and the next argument

File: test/methods.js (hunk 5 above)
Comment (grawk) [inline test/methods.js]: whitespace, please put a space between a comma and the next argument

File: lib/locreator.js [hunk 8]
```diff
@@ -75,6 +75,12 @@ Locreator.prototype.addStarMethods = function locreator(locatorId, _locator, par
   locatorObject[locatorId + 'OptionValue'] = function (optionValue) {
     return drivex.selectByOptionValue(locator, optionValue, parentWebElement);
   };
+  locatorObject[locatorId + 'ValidateText'] = function (ValidateText) {
+    return drivex.validateText(locator, parentWebElement,ValidateText) ;
```
Comment (grawk) [inline lib/locreator.js]: whitespace, please put a space between a comma and the next argument

File: lib/locreator.js (hunk 2 above)
Comment (grawk) [inline lib/locreator.js]: whitespace, please put a space between a comma and the next argument

Comment (grawk) [inline lib/locreator.js]: whitespace, please put a space between a comma and the next argument

File: lib/locreator.js [hunk 9]
```diff
@@ -75,6 +75,12 @@ Locreator.prototype.addStarMethods = function locreator(locatorId, _locator, par
...
+  };
+  locatorObject[locatorId + 'ValidateAttributeValue'] = function (attribute,validateText) {
+    return drivex.validateAttributeValue(locator, parentWebElement,attribute,validateText);
```
Comment (grawk) [inline lib/locreator.js]: whitespace, please put a space between a comma and the next argument

File: lib/locreator.js (hunk 1 above)
Comment (grawk) [inline lib/locreator.js]: function argument should begin with lowercase letter

Comment (grawk) [inline lib/locreator.js]: can you change the name of this argument to `comparisonText` or something else that succinctly describes its purpose?

File: lib/locreator.js (hunk 2 above)
Comment (grawk) [inline lib/locreator.js]: can you please change the name of the parameter  from `validateText` to `comparisonValue` or `value` or something else which succinctly describes the purpose?

Comment (nikulkarni) [inline lib/locreator.js]: I like `AttributeEquals`

File: lib/locreator.js (hunk 1 above)
Comment (nikulkarni) [inline lib/locreator.js]: +1 for `TextEquals`

Comment (vidyaravind) [inline lib/locreator.js]: Changing name to TextEquals and AttrEquals

File: README.md [hunk 10]
```diff
@@ -362,6 +362,19 @@ Any method in the view object's prototype will also be available for use
 Any method in the view object's prototype will also be available for use
 Other than that, the nemo-view uses nemo-locatex internally, so if you change your locator files and set LOCALE, nemo-view will handle the rest!
 
+#### [locatorName]TextEquals
+
+* arguments
+  * text: the expected text on the element
```
Comment (grawk) [inline README.md]: in the code the parameter is `value` so it should match in the doc

File: README.md [hunk 11]
```diff
@@ -362,6 +362,19 @@ Any method in the view object's prototype will also be available for use
...
+* returns: Promise which resolves to true when the expected text matches
+
+#### [locatorName]AttrEquals
+
+* arguments
+  * text: attribute value to check
```
Comment (grawk) [inline README.md]: in the code the first argument is `attribute` not `text`

File: README.md [hunk 12]
```diff
@@ -362,6 +362,19 @@ Any method in the view object's prototype will also be available for use
...
+  * text: the expected text on the element
```
Comment (grawk) [inline README.md]: in the code the second argument is `value` not `text`

**Pull request merged**